src/reforgeClient.ts

```diff
@@ -0,0 +1,110 @@
+import { contextObjToMap, mergeContexts } from "./mergeContexts";
+import type {
+  Contexts,
+  ContextObj,
+  LogLevel,
+  ReforgeInterface,
+  Telemetry,
+  TypedNodeServerConfigurationRaw,
+} from "./types";
+import type { makeLogger } from "./logger";
+import type { GlobalListenerCallback } from "./configChangeNotifier";
+
+/**
+ * A context-scoped Reforge client that shares the underlying resolver
+ * with its parent Reforge instance but applies a specific context to all operations.
+ *
+ * This allows creating request/call-scoped clients without duplicating
+ * the resolver, polling, or SSE connections. All scoped clients share
+ * the same parent Reforge's state.
+ */
+export class ReforgeClient implements ReforgeInterface {
+  private readonly parent: ReforgeInterface;
+  private readonly context: Contexts;
+
+  constructor(parent: ReforgeInterface, contexts: Contexts | ContextObj) {
+    this.parent = parent;
+    this.context =
+      contexts instanceof Map ? contexts : contextObjToMap(contexts);
+  }
+
+  get telemetry(): Telemetry | undefined {
```

Comment:
Does this actually need to be exposed?

Reply:
the only reason i could see exposing this is when debugging telemetry or wanting to force a flush. I think that probably means it ought not be exposed.